src/frontend/web_application/src/components/OpenPGPKey/index.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { Component, PropTypes } from 'react';
+import { enums } from 'openpgp';
+import classnames from 'classnames';
+import { DateTime, withTranslator } from '@gandi/react-translate';
+import Button from '../Button';
+import Icon from '../Icon';
+import DefList from '../DefList';
+import { getKeyFromASCII } from '../../services/openpgp-manager';
+import './style.scss';
+
+function getViewValues(publicKey, keyStatuses) {
+  return {
+    fingerprint: publicKey.primaryKey.fingerprint,
+    userId: publicKey.getPrimaryUser().user.userId.userid,
+    createdAt: publicKey.primaryKey.created,
+    expirationTime: publicKey.getExpirationTime(),
+    algorithm: publicKey.primaryKey.algorithm,
+    bitSize: publicKey.primaryKey.getBitSize(),
+    userIds: publicKey.users.map(user => user.userId.userid),
+    keyStatus: Object.keys(keyStatuses)
+      .find(statusLiteral => keyStatuses[statusLiteral] === publicKey.verifyPrimaryKey()),
+  };
+}
+
+@withTranslator()
+class OpenPGPKey extends Component {
+  static propTypes = {
+    __: PropTypes.func,
+    children: PropTypes.node,
+    publicKeyArmored: PropTypes.string.isRequired,
+    privateKeyArmored: PropTypes.string,
+    editMode: PropTypes.bool,
+    onDeleteKey: PropTypes.func,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      showDetails: false,
+    };
+
+    this.toggleDetails = this.toggleDetails.bind(this);
+    this.handleDeleteKey = this.handleDeleteKey.bind(this);
+    this.keyStatus = enums.keyStatus;
```

Comment:
I was not proud of that; I always keep in mind the quote from Head first design pattern book "program an interface, not an implementation" but since there are no dependency injection inside react components it is just weird.

Well I fixed with @themouette reviews.